src/ray/core_worker/core_worker.cc

```diff
@@ -248,6 +259,19 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
   }
 }
 
+void CoreWorker::HandleWorkerFailed(const WorkerID &worker_id,
```

Comment:
Are you planning to do any of this cleanup in this PR? If not, is there some way you can test the publishing logic on its own?

Reply:
I just tested manually by checking the debug messages. Is that okay? If not, I can also implement the fate sharing in this PR.